### PR TITLE
Fixes a few roulette wheel runtimes.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Roulette wheel would runtime/bluescreen when an observer used it because of a naked cast of user to /mob/living/carbon/human
Roulette wheel would runtime/bluescreen when the ID card linked to it lost its registered account for any reason.
Roulette wheel would runtime/bluescreen when the user had an ID card that had no registered account.

Fixes all three of these runtimes.

## Why It's Good For The Game

Runtimes bad.

## Changelog
:cl:
fix: The Roulette Wheel interface should no longer bluescreen for bank account-less ID cards and observers.
/:cl:

### CODE REVIEW

**Comment by notfrying1pans** [comment]
thanks :)

File: code/game/machinery/roulette_machine.dm [hunk 1]
```diff
@@ -71,16 +71,19 @@
 	data["IsAnchored"] = anchored
 	data["BetAmount"] = chosen_bet_amount
 	data["BetType"] = chosen_bet_type
-	data["HouseBalance"] = my_card?.registered_account.account_balance
+	data["HouseBalance"] = my_card?.registered_account ? my_card?.registered_account.account_balance : 0
```
Comment (Mothblocks) [inline code/game/machinery/roulette_machine.dm:74]:
```suggestion
	data["HouseBalance"] = my_card?.registered_account?.account_balance || 0
```

File: code/game/machinery/roulette_machine.dm [hunk 2]
```diff
@@ -71,16 +71,19 @@
...
-	var/mob/living/carbon/human/H = user
-	var/obj/item/card/id/C = H.get_idcard(TRUE)
-	if(C)
-		data["AccountBalance"] = C.registered_account.account_balance
+
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/human_user = user
+		var/obj/item/card/id/id_card = human_user.get_idcard(TRUE)
+		data["AccountBalance"] = id_card?.registered_account ? id_card.registered_account.account_balance : 0
```
Comment (Mothblocks) [inline code/game/machinery/roulette_machine.dm:82]:
```suggestion
		data["AccountBalance"] = id_card?.registered_account?.account_balance || 0
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Two nit-picks to create more readable and maintainable code (since it'll ensure only one place needs to be touched, not two).

**Pull request merged**